### ISSUE TITLE
7.14.x/7.12.x Dropdown default value overwrites saved value in editview

### ISSUE THREAD

**Issue by gunnicom** [salesagility/SuiteCRM#10174]
#### Issue
If a dropdown ist defined with '' (empty) as possible value, and a default value that is any other value following happesn:
Save an object with value of ''. In Detailview you will see the value saved correctly. Open in Detailview and you will not see the saved '' but the default value. If you save now, you save the default value.

I think it is critical, because you may not have the field in your view, leading to change a value without knowing to a value that is not intended. 
For me it is nessecary to save an empty value sometimes, and it seems unlogical that you can define an empty value and then it gets overwritten.

#### Expected Behavior
Do not change empty values to default value in editview

#### Actual Behavior
Suite changes value to default value.

#### Steps to Reproduce
1. Define a field as dropdown with one choice being blank '', and some other choices. Example:
```
$app_list_strings['test_list']=array(
  '' => '',
  'a1' => 'a1',
  'b2' => 'b2',
  'c3' => 'c3',
  );
```
2. Chose b2 as default
3. save an object with dropdown value choosen is ''
4. View in Detailview: '' gets shown correctly
5. Go to EditView or InlineEdit field, field gets value 'b2'

#### Context
I think it is critical to change data with possible not even knowing it gets changed.
For me it is a legitimate scenario to save '' in an dropdown also default may be different.

#### Your Environment

* SuiteCRM Version used: Tested on 7.14.0 and 7.12.0
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Chrome Firefox
* Environment name and version (e.g. MySQL, PHP 7): PHP 7.4 and 8.2
* Operating System and version (e.g Ubuntu 16.04): Ubuntu 22.04

**Comment by timo-ecm2** [comment]
Yes I can confirm this with Version 7.12.12 -Sugar Version 6.5.25 (Build 344) in combination wth Chrome - Version 116.0.5845.180 (Officiële build) (64-bits).
